rockcraft/oci.py

```diff
@@ -355,17 +355,28 @@ def set_default_user(self, user: str) -> None:
         _config_image(image_path, params)
         emit.progress(f"Default user set to {user}")
 
-    def set_entrypoint(self) -> None:
+    def set_entrypoint(
+        self, entrypoint_service: Optional[str], args: Optional[str]
+    ) -> None:
         """Set the OCI image entrypoint. It is always Pebble and CMD is null."""
         emit.progress("Configuring entrypoint...")
+        cmd_params = ["--clear=config.cmd"]
         image_path = self.path / self.image_name
         entrypoint = [f"/{Pebble.PEBBLE_BINARY_PATH}", "enter", "--verbose"]
+        if entrypoint_service:
+            if args:
+                parts = args.split()
+            else:
+                parts = []
+            entrypoint.extend(["--args", entrypoint_service])
```

Comment:
this is the only line that matters in this function. the rest should be moved to a new function. See my explanation below in https://github.com/canonical/rockcraft/pull/410/files#diff-cf13f6f606f30afa23cf9954776527355fc805a6d81f60f29ec6899c261d83e9R172